packages/interactivity/src/hooks.tsx

```diff
@@ -260,18 +261,35 @@ export const directive = (
 
 // Resolve the path to some property of the store object.
 const resolve = ( path, namespace ) => {
+	if ( namespace === '' ) {
+		if ( isDebug ) {
+			// eslint-disable-next-line no-console
+			console.warn(
+				`Namespace cannot be an empty string. Error found when trying to use "${ path }"`
```

Comment:
I wonder if there is a use case for an island that wants to hydrate but doesn't have a namespace, and all its directives use the custom namespace syntax.

Anyway, we can add this and then remove it later if necessary 👍🙂

Reply:
I was worried about that myself and did some testing. This check seems to fire with the "resolved" namespace, so this warns:

```html
<div data-wp-interactive data-wp-text='state.x'></div>
```

But this is fine:

```html
<div data-wp-interactive data-wp-text='explicitNS::state.x'></div>
```

@cbravobernal if we don't have a test like this can we add it? It seems perfectly legitimate to have an interactive region without its own namespace.

Reply:
I think these won't warn (see https://github.com/WordPress/gutenberg/pull/61409/files#r1594570115):

```html
<div data-wp-interactive='null' data-wp-text='state.x'></div>
<div data-wp-interactive='{}' data-wp-text='state.x'></div>
```

Reply:
Good call @DAreRodz - I think we should revisit this section to fix some of that - we JSON.parse attribute values too aggressively:

https://github.com/WordPress/gutenberg/blob/78e53bb99f7e0bf77eb2e0c4576358c83f3b1d72/packages/interactivity/src/vdom.ts#L82-L97

Maybe something like this (only for the `interactive` directive):

- If the value is falsy => null ns
- If the value starts with `{` (I don't think namespaces are allowed to start with this?)
  - Try to parse and get the namespace property
  - otherwise `null`
- Otherwise, the namespace is the string - should we check that it's a valid namespace??

I don't _think_ that any other directives should ever be JSON parsed.

Reply:
> I don't think that any other directives should ever be JSON parsed.

What about `data-wp-context`?

Reply:
```
<div data-wp-interactive='null' data-wp-text='state.x'></div>
<div data-wp-interactive='{}' data-wp-text='state.x'></div>
```
Now they should.

> Maybe something like this (only for the interactive directive)....

Should we do this in a follow-up PR?

Reply:
Yep, let's leave those refactors for another PR.

> What about data-wp-context?

🙃